spinn_front_end_common/interface/splitter_selectors/splitter_selector.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+
+from pacman.model.partitioner_splitters import (
+    SplitterOneAppOneMachine, SplitterOneToOneLegacy, SplitterSliceLegacy)
+from spinn_front_end_common.utility_models import (
+    ReverseIpTagMultiCastSource, LivePacketGather, ChipPowerMonitor)
+from pacman.model.graphs.application.abstract import (
+    AbstractOneAppOneMachineVertex)
+from spinn_utilities.log import FormatAdapter
+
+logger = FormatAdapter(logging.getLogger(__name__))
+
+
+class SplitterSelector(object):
+    """ splitter object selector that allocates nothing but legacy
+    splitter objects where required
+    """
+
+    NOT_KNOWN_APP_VERTEX_ERROR_MESSAGE = (
+        "The SplitterSelector has not seen the {} vertex before. Therefore "
+        "there is no known splitter to allocate to this app vertex and so "
+        "will use the SplitterSliceLegacy Splitter.")
+
+    def __call__(self, app_graph):
+        """ basic selector which puts the legacy splitter object on
+        everything without a splitter object
+
+        :param ApplicationGraph app_graph: app graph
+        :rtype: None
+        """
+        for app_vertex in app_graph:
+            if app_vertex.splitter is None:
+                self.vertex_selector(app_vertex)
+
+    def vertex_selector(self, app_vertex):
+        """ main point for selecting a splitter object for a given app vertex.
```

Comment:
Backslash needed at end of every non-terminal line in this paragraph _or_ blank line before next line. I think a blank line is probably better in this case.

Reply:
fixed